components/user-mgt/org.wso2.carbon.user.mgt/src/main/java/org/wso2/carbon/user/mgt/listeners/UserManagementAuditLogger.java

```diff
@@ -430,4 +498,36 @@ private void addContextualAuditParams(JSONObject jsonObject) {
         jsonObject.put(USER_NAME_KEY, MDC.get(USER_NAME_QUERY_KEY));
         jsonObject.put(SERVICE_PROVIDER_KEY, MDC.get(SERVICE_PROVIDER_QUERY_KEY));
     }
+
+    /**
+     * Mask claims in audit logs based on the masking config.
+     *
+     * @param claims    Claims map.
+     * @param data      JSON Object which will be added to audit log.
+     */
+    private void maskClaimsInAuditLog(Map<String, String> claims, JSONObject data){
+
+        if (LoggerUtils.isLogMaskingEnable) {
+            Map<String, String> maskedClaims = LoggerUtils.getMaskedClaimsMap(claims);
+            data.put(ListenerUtils.CLAIMS_FIELD, new JSONObject(maskedClaims));
+        } else {
+            data.put(ListenerUtils.CLAIMS_FIELD, new JSONObject(claims));
+        }
+    }
+
+    /**
+     * Returns the target value based on the masking config.
+     *
+     * @param userName          Claims map.
+     * @param userStoreManager  JSON Object which will be added to audit log.
+     * @return masked target value.
+     */
+    public static String getTargetForAuditLog(String userName, UserStoreManager userStoreManager) {
+
+        String target = ListenerUtils.getEntityWithUserStoreDomain(userName, userStoreManager);
```

Comment:
this is not needed if log masking is enabled. hence let's improve the code to avoid of the execution of this in such cases

Reply:
if log masking is enabled, this value will be masked. Therefore in both scenarios this is needed.